src/Validator/Rules/ProvidedRequiredArguments.php

```diff
@@ -19,7 +20,7 @@ public function getVisitor(ValidationContext $context)
     {
         return [
             NodeKind::FIELD     => [
-                'leave' => static function (FieldNode $fieldNode) use ($context) {
+                'leave' => static function (FieldNode $fieldNode) use ($context) : ?VisitorOperation {
                     $fieldDef = $context->getFieldDef();
 
                     if (! $fieldDef) {
```

Comment:
I'd invert the condition

Reply:
I don't disagree, but that sort of opens a can of worms and I think that kind of thing will be covered by other phpstan warnings which I'll tackle in a new PR; I'd rather focus on "return native typehint" errors in this PR if you don't mind.